code/game/objects/structures/tank_dispenser.dm

```diff
@@ -88,13 +88,13 @@
 	switch(action)
 		if("plasma")
 			var/obj/item/weapon/tank/internals/plasma/tank = locate() in src
-			if(tank)
+			if(tank && Adjacent(usr))
 				usr.put_in_hands(tank)
 				plasmatanks--

```

Comment:
No, why should it go to the floor if the user is not adjacent? It should not do anything at all